src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -113,51 +120,252 @@ def checkLastImage(stage) {
+ * the logs produced from executing a task to artifacts.ci.centos.org
+ *
+ * variables
+ *  currentStage - current stage running
+ */
+def rsyncData(currentStage){
+    def text = "export JENKINS_JOB_NAME=\"${env.JOB_NAME}-${currentStage}\"\n" +
+            "export RSYNC_USER=\"${env.RSYNC_USER}\"\n" +
+            "export RSYNC_SERVER=\"${env.RSYNC_SERVER}\"\n" +
+            "export RSYNC_DIR=\"${env.RSYNC_DIR}\"\n" +
+            "export FEDORA_PRINCIPAL=\"${env.FEDORA_PRINCIPAL}\"\n" +
+            "export JENKINS_BUILD_TAG=\"${env.BUILD_TAG}-${currentStage}\"\n" +
+            "export OSTREE_BRANCH=\"${env.OSTREE_BRANCH}\"\n"
+
+    if (currentStage in ['ci-pipeline-ostree-compose', 'ci-pipeline-ostree-iamge-compose',
+                         'ci-pipeline-ostree-image-boot-sanity', 'ci-pipeline-ostree-boot-sanity']) {
+        text = text +
+                "export HTTP_BASE=\"${env.HTTP_BASE}\"\n" +
+                "export branch=\"${env.branch}\"\n"
+    }
+    if (currentStage == 'ci-pipeline-rpmbuild') {
+        text = text +
+                "export fed_repo=\"${env.fed_repo}\"\n" +
+                "export fed_branch=\"${env.fed_branch}\"\n" +
+                "export fed_rev=\"${env.fed_rev}\"\n"
+
+    } else if (currentStage == 'ci-pipeline-ostree-image-boot-sanity') {
+        text = text +
+                "export ANSIBLE_HOST_KEY_CHECKING=\"False\"\n"
+    } else if (currentStage == 'ci-pipeline-ostree-boot-sanity') {
+        text = text +
+                "export fed_repo=\"${fed_repo}\"\n" +
+                "export image2boot=\"${image2boot}\"\n" +
+                "export commit=\"${commit}\"\n" +
+                "export ANSIBLE_HOST_KEY_CHECKING=\"False\"\n"
+    }
+
+    writeFile file: "${env.ORIGIN_WORKSPACE}/task.env",
+            text: text
+    runTaskAndReturnLogs(currentStage, 'duffy-key')
+
+}
+
+/**
+ * Library to provision resources used in the current stage
+ *
+ * variables
+ *  currentStage - current stage running
+ */
+def provisionResources(currentStage){
+
+    utils.allocateDuffyCciskel()
+
+    echo "Duffy Allocate ran for stage ${currentStage} with option --allocate\r\n" +
+            "ORIGIN_WORKSPACE=${env.ORIGIN_WORKSPACE}\r\n" +
+            "ORIGIN_BUILD_TAG=${env.ORIGIN_BUILD_TAG}\r\n" +
+            "ORIGIN_CLASS=${env.ORIGIN_CLASS}"
+
+    job_props = "${env.ORIGIN_WORKSPACE}/job.props"
+    job_props_groovy = "${env.ORIGIN_WORKSPACE}/job.groovy"
+    convertProps(job_props, job_props_groovy)
+    load(job_props_groovy)
+
+}
+
+/**
+ * Library to teardown resources used in the current stage
+ *
+ * variables
+ *   currentStage - current stage running
+ */
+def teardownResources(currentStage){
+
+    utils.teardownDuffyCciskel()
+
+    echo "Duffy Deallocate ran for stage ${currentStage} with option --teardown\r\n" +
+            "DUFFY_HOST=${env.DUFFY_HOST}"
+}
+
+
+def convertProps(file1, file2) {
```

Comment:
This method is already in the cico-pipeline-library

Reply:
Ack, will remove from PIpelineUtils

Reply:
Maybe it is better to make the changes there?  Or rename the method to something else then what is in the cico-pipeline-library.   It would be beneficial to update general libraries there and more specific libraries for our work in our repo IMHO